RELEASE_PROCESS.md

```diff
@@ -0,0 +1,51 @@
+# Release Process
+
+This document contains all the necessary information about out release process.
+
+## Release branch
+
+For each release we create a separate release branch.
+
+Commits / Features which should be included in the release are moved over into the release branch (e.g. cherry-picked).
+
+We open up a separate PR for the release branch (and add the PR to the milestone) to track the current status.
```

Comment:
And this PR would include ALL the changes of the release? I'm guessing we're not gonna review that as it's a HUGE PR?